src/module.ts

```diff
@@ -0,0 +1,93 @@
+import { resolve } from 'path'
+import { fileURLToPath } from 'url'
+import { defineNuxtModule, addPlugin, isNuxt2 } from '@nuxt/kit'
+import defu from 'defu'
+
+import { name, version } from '../package.json'
+
+import logger from './logger'
+
+export interface ModuleOptions {
+  id?: string,
+  dev?: boolean,
+  metrikaUrl?: string,
+  accurateTrackBounce?: boolean | number,
+  childIframe?: boolean,
+  clickmap?: boolean,
+  defer?: boolean,
+  ecommerce?: boolean | string | [],
+  params?: object | [],
+  useCDN?: boolean,
+  userParams?: object,
+  trackHash?: boolean,
+  trackLinks?: boolean,
+  trustedDomains?: [],
+  type?: number,
+  webvisor?: boolean,
+  triggerEvent?: boolean,
+}
+
+const CONFIG_KEY = 'yandexMetrika'
+
+declare module '@nuxt/schema' {
+  interface PublicRuntimeConfig {
+    [CONFIG_KEY]?: ModuleOptions
+  }
+}
+
+export default defineNuxtModule<ModuleOptions>({
+  meta: {
+    name,
+    version,
+    configKey: CONFIG_KEY,
+    compatibility: {
+      bridge: true
+    }
+  },
+  defaults: {
+    metrikaUrl: 'https://mc.yandex.ru/metrika',
+    accurateTrackBounce: true,
+    childIframe: false,
+    clickmap: true,
+    defer: false,
+    trackHash: false,
+    trackLinks: true,
+    type: 0,
+    webvisor: false,
+    triggerEvent: false
+  },
+  setup (options: ModuleOptions, nuxt) {
+    // Don't include on dev mode
+    if (options.dev && process.env.NODE_ENV !== 'production') {
+      return
+    }
+
+    if (!options.id) {
+      logger.error('No id provided.')
```

Comment:
Can we have a way to disable/not enable metrika in runtime without logging an error message? Such as for running e2e tests against the production build.

Reply:
I'm not sure how is this resolved? Again, I'd like to be able to run `YANDEX_METRIKA_ID=xxx nuxt start` and have metrika enabled, then run `nuxt start` and have metrika script not loaded. I suppose with the current implementation that is going to be possible but will emit this error?